src/common/websocket/api.ts

```diff
@@ -0,0 +1,101 @@
+import { getAppId, getSocketURL } from './config'
+import { getCrowdin, getLanguage } from 'common/utility'
+import { brand_name } from 'common/constants'
+
+interface DerivAPIRequest {
+    passthrough?: {
+        [k: string]: unknown
+    }
+    req_id?: number
+    [k: string]: unknown
+}
+
+interface DerivAPIResponse {
+    echo_req: {
+        [k: string]: unknown
+    }
+    req_id?: number
+    [k: string]: unknown
+}
+
+class CustomPromise<T> {
+    public readonly promise: Promise<T>
+    private resolveFn!: (value: T | PromiseLike<T>) => void
+    private rejectFn!: (reason?: any) => void
+
+    public constructor() {
+        this.promise = new Promise<T>((resolve, reject) => {
+            this.resolveFn = resolve
+            this.rejectFn = reject
+        })
+    }
+
+    public reject(reason?: any) {
+        this.rejectFn(reason)
+    }
+
+    public resolve(param: T) {
+        this.resolveFn(param)
+    }
+}
+
+export default class DerivAPI {
+    private pendingRequests: {
+        [k: string]: CustomPromise<DerivAPIResponse>
+    } = {}
+    private reqId = 0
+    private connected = new CustomPromise<void>()
+    connection: WebSocket
+
+    constructor() {
+        const socket_url = `${getSocketURL()}?app_id=${getAppId()}&l=${
+            getLanguage() === 'ach' ? getCrowdin() : getLanguage()?.replace('-', '_')
+        }&brand=${brand_name.toLowerCase()}`
+
+        const connection = new WebSocket(socket_url)
+        connection.onopen = this.openHandler
+        connection.onmessage = this.messageHandler
+
+        this.connection = connection
+    }
+
+    isConnectionClosed() {
+        return this.connection.readyState === 2 || this.connection.readyState === 3
```

Comment:
`this.connection.readyState === 2`
For these numbers can we save them as a meaningfull variable name and use it to compare